account_payment_cancel/tests/test_payment_cancel.py

```diff
@@ -0,0 +1,145 @@
+from odoo.tests import TransactionCase
+from odoo.modules import get_module_resource
+
+
+class TestPaymentCancel(TransactionCase):
+
+    def setUp(self):
+        super(TestPaymentCancel, self).setUp()
+
+        self.invoice_name = 'test invoice pain000'
+        self.invoice_line_name = 'test invoice line pain000'
+        self.order_name = '2017/1013'
+        self.journal_name = '2017/1013'
+        self.payment_line_name = 'Ltest'
+
+        # Create payment order from the invoice
+        invoice = self.env['account.invoice'].search(
+            [('move_name', '=', self.invoice_name)])
+
+        # Validate the invoice
+        invoice.action_invoice_open()
+
+        # Create a payment order
+        action = invoice.create_account_payment_line()
+        payment_order_id = action['res_id']
+
+        payment_order = self.env['account.payment.order'].search(
+            [('id', '=', payment_order_id)])
+
+        partner_bank = self.env['account.journal'].search(
+            [('name', '=', self.journal_name)])
+
+        bank = self.env['account.journal'].search([('name', '=', 'Bank')])
+
+        payment_order.name = self.order_name
+        bank.update_posted = True
+
+        payment_order.journal_id = partner_bank.id
+        # Confirm payment order
+        payment_order.draft2open()
+        # Generate payment file
+        payment_order.open2generated()
+        # File successfully uploaded
+        payment_order.generated2uploaded()
+
+        payment_line = self.env['bank.payment.line'].search(
```

Comment:
You can directly store your payment_line in self.payment_line to have access to it in the tests.